Code/Framework/AzToolsFramework/AzToolsFramework/Manipulators/PaintBrushManipulator.cpp

```diff
@@ -89,30 +45,39 @@ namespace AzToolsFramework
         const AZ::Transform& worldFromLocal, const AZ::EntityComponentIdPair& entityComponentIdPair)
     {
         m_ownerEntityComponentId = entityComponentIdPair;
-        m_config.m_ownerEntityComponentId = entityComponentIdPair;
-        PaintBrushRequestBus::Handler::BusConnect(entityComponentIdPair);
 
         SetSpace(worldFromLocal);
 
+        // Make sure the Paint Brush Settings window is open
+        AzToolsFramework::OpenViewPane(PaintBrush::s_paintBrushSettingsName);
```

Comment:
question: Should this be part of the Component Mode as opposed to the manipulator itself? Or are they closely enough related they should always exist together?

Reply:
They're closely related. PaintBrushSettings / PaintBrushManipulator are the core of the paint brush tool. The general idea here is that I want PaintBrushManipulator to contain all of the common logic for paint brushes, so each new type of component that needs a paintbrush (vertex weight painting, texture painting, whitebox, whatever) would just need to add the PaintBrushManipulator and have a thin component mode that ties the general concept of "painting" to the specific data manipulation needed for that component type. If I move logic for paint brush settings etc up to the component mode level, then it's going to get duplicated / varied for each of those component modes, or else require a base-class component mode, or something. I could easily be talked into the base-class component mode route instead of what I have here, but that would be in a future iteration I think, and ideally wouldn't happen until we had a second consumer, so that the design of that base class was better understood.

Reply:
So one thing to note is it's possible to have a Component Mode be shared across multiple different Component types if you want (though this can be difficult), we do this for Box Shape Component Mode and Box Collider Component Mode. The other approach is to have the Component Mode for specific types be minimal, and delegate to a `PaintBrushMode` type (separate from the manipulator) that the lions share of the logic is in. That way logic isn't duplicated across different Component Modes (they just call into `PaintBrushMode`) and `PaintBrushMode` manages the Manipulator logic only. It's a small distinction but just keeps things smaller and separated which can make testing easier (say if you want to just test the input of `PaintBrushManipulator` without all the Component Mode logic getting tied into it. It's not a huge deal but just a suggestion that might be useful in future as the design evolves.